File: lab_12/replace_word.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main() {
    ifstream file("data.txt");
    vector<string> lines;

    if (!file) {
        cerr << "Unable to open file data.txt. Creating a new file with default values." << endl;
        vector<string> default_data = {"Hello", "World", "C++", "first initial"};
        ofstream outfile("data.txt");
        for (const auto &l : default_data) {
            outfile << l << endl;
        }
        outfile.close();
        lines = default_data;
    } else {
        string line;
        while (getline(file, line)) {
            lines.push_back(line);
        }
        file.close();
    }

    string old_word, new_word;
    cout << "Enter word to replace: ";
    cin >> old_word;
    cout << "Enter new word: ";
    cin >> new_word;

    for (auto &l : lines) {
        if (l == old_word) {
            l = new_word;
        }
    }

    ofstream outfile("data.txt");
    for (const auto &l : lines) {
        outfile << l << endl;
    }
    outfile.close();

    cout << "[แก้ไขข้อมูลในไฟล์เรียบร้อย!]" << endl;
    return 0;
}

File: lab_12/replace_word.py
def replace_word_in_file(file_path, old_word, new_word):
    with open(file_path, 'r') as file:
        lines = file.readlines()

    with open(file_path, 'w') as file:
        for line in lines:
            if line.strip() == old_word:
                file.write(new_word + '\n')
            else:
                file.write(line)

if __name__ == "__main__":
    file_path = 'data.txt'
    old_word = input("Enter word to replace: ")
    new_word = input("Enter new word: ")
    replace_word_in_file(file_path, old_word, new_word)
    print("[แก้ไขข้อมูลในไฟล์เรียบร้อย!]")
